keyboards/pica40/rev2/rev2.c

```diff
@@ -0,0 +1,200 @@
+// Copyright 2022 zzeneg (@zzeneg)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "rev2.h"
+
+#ifdef ENCODER_ENABLE // code based on encoder.c
+
+static const pin_t encoders_pad_a[] = ENCODERS_PAD_A;
+static const pin_t encoders_pad_b[] = ENCODERS_PAD_B;
+
+static int8_t  encoder_LUT[]  = {0, -1, 1, 0, 1, 0, 0, -1, -1, 0, 0, 1, 0, 1, -1, 0};
+static uint8_t encoder_state  = 3;
+static int8_t  encoder_pulses = 0;
+static uint8_t encoder_value  = 0;
+
+typedef struct encoder_sync_data {
+    int value;
+} encoder_sync_data;
+
+// custom handler that returns encoder B pin status from slave side
```

Comment:
why? 

In theory, you'd need to read both pins from the same side since the encoder pins need to be connected on that half.

Reply:
That's the thing - I have only 1 free pin on each MCU, so my encoder is connected to both halves, A pin to master and B to slave (through TRRS).

Reply:
what happens if one half is disconnected?

it may be worth checking `is_transport_connected()` before reading from the split pin to make sure it's even connected. That should help reduce misreads

Reply:
Actually `transaction_rpc_recv` already calls `is_transport_connected` at the beginning, so I don't think it's needed.